From fb5b1b9bd29db98720add53edaff76cb46e958a2 Mon Sep 17 00:00:00 2001
From: Miroslaw Swiader <miroslaw.swiader@intel.com>
Date: Mon, 8 Feb 2016 12:49:08 +0100
Subject: [PATCH 2/6] usb: dwc3: Fix for disable SUSPHY is applicable only for
 USB2.0.

According to Synopsys spec SUSPHY bit should
be set only for USB2.0 speed.

Tracked-On: https://jira01.devtools.intel.com/browse/OAM-19339
Change-Id: Idaeb90453f324f5da872f12ffd8048b5fd797d18
Signed-off-by: Miroslaw Swiader <miroslaw.swiader@intel.com>
Reviewed-on: https://android.intel.com:443/467347
---
 drivers/usb/dwc3/gadget.c | 14 +++++++++-----
 1 file changed, 9 insertions(+), 5 deletions(-)

diff --git a/drivers/usb/dwc3/gadget.c b/drivers/usb/dwc3/gadget.c
index 97fb11d..fbb05c5 100644
--- a/drivers/usb/dwc3/gadget.c
+++ b/drivers/usb/dwc3/gadget.c
@@ -322,7 +322,7 @@ int dwc3_send_gadget_ep_cmd(struct dwc3 *dwc, unsigned ep,
 	struct dwc3_ep		*dep = dwc->eps[ep];
 	u32			timeout = 500;
 	u32			reg;
-	u32			phycfg_val;
+	u32			phycfg_val = 0;
 
 	dev_vdbg(dwc->dev, "%s: cmd '%s' params %08x %08x %08x\n",
 			dep->name,
@@ -333,10 +333,12 @@ int dwc3_send_gadget_ep_cmd(struct dwc3 *dwc, unsigned ep,
 	dwc3_writel(dwc->regs, DWC3_DEPCMDPAR1(ep), params->param1);
 	dwc3_writel(dwc->regs, DWC3_DEPCMDPAR2(ep), params->param2);
 
-	phycfg_val = dwc3_readl(dwc->regs, DWC3_GUSB2PHYCFG(0));
+	if (dwc->gadget.speed != USB_SPEED_SUPER) {
+		phycfg_val = dwc3_readl(dwc->regs, DWC3_GUSB2PHYCFG(0));
 
-	dwc3_writel(dwc->regs, DWC3_GUSB2PHYCFG(0),
-		(phycfg_val & ~DWC3_GUSB2PHYCFG_SUSPHY));
+		dwc3_writel(dwc->regs, DWC3_GUSB2PHYCFG(0),
+			(phycfg_val & ~DWC3_GUSB2PHYCFG_SUSPHY));
+	}
 
 	dwc3_writel(dwc->regs, DWC3_DEPCMD(ep), cmd | DWC3_DEPCMD_CMDACT);
 	do {
@@ -344,7 +346,9 @@ int dwc3_send_gadget_ep_cmd(struct dwc3 *dwc, unsigned ep,
 		if (!(reg & DWC3_DEPCMD_CMDACT)) {
 			dev_vdbg(dwc->dev, "Command Complete --> %d\n",
 					DWC3_DEPCMD_STATUS(reg));
-			dwc3_writel(dwc->regs, DWC3_GUSB2PHYCFG(0), phycfg_val);
+			if (dwc->gadget.speed != USB_SPEED_SUPER)
+				dwc3_writel(dwc->regs, DWC3_GUSB2PHYCFG(0),
+				phycfg_val);
 			if (DWC3_DEPCMD_STATUS(reg))
 				return -EINVAL;
 
-- 
1.9.1

